docs/api/connection.md

```diff
@@ -105,7 +105,7 @@ Optional
 
 > Default: `{}`
 
-> `options.results` -- Object[]
+> `options.results` -- [`Doc`]({{ site.baseurl }}{% link api/doc.md %})[]
```

Comment:
Very nice! It might help to explicitly state in the docs what is expected with regards to subscription status.

Reply:
I agree — might be nice to expand on how to actually get this `Doc` array (I have to admit I've never done this, so when I wrote the documentation, I didn't add any details)

Reply:
Cool! I'm (hopefully) going to be experimenting with this API soon, so I can test out what happens in the various cases and report back / update the docs when I do:
 * Doc not fetched or subscribed
 * Doc fetched but not subscribed
 * Doc fetched and subscribed

Reply:
I have verified the behavior for the various cases.

Short answer: any combination of fetch and subscribe states appears to work, as long as each doc in `results` is initialized using `connection.get`.

Long answer:
 * If the docs in `results` are initialized with `connection.get` followed by `ingestSnapshot` (without subscribing), then the query implementation will subscribe to all of them (note that you still need to manually add the listeners for `doc.on('load')` and `doc.on('op')` - the subscribeQuery does not fire events when the individual docs are updated, only when the query result list is changed). In this case, we get a message that looks like this over WebSockets `a: [["id1", 10], ["id2", 3],...`. This does not contain the full documents, only their ids and versions, which is the expected behavior here. Awesome!
 * If the docs in `results` are initialized with only `connection.get` and nothing else, then a message appears overWebSockets that contains the full document contents for each individual doc.